de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/picking/candidate/commands/CreatePickingCandidatesCommand.java

```diff
@@ -0,0 +1,100 @@
+package de.metas.handlingunits.picking.candidate.commands;
+
+import org.adempiere.ad.trx.api.ITrxManager;
+import org.compiere.model.I_C_UOM;
+
+import de.metas.handlingunits.HuId;
+import de.metas.handlingunits.model.I_M_ShipmentSchedule;
+import de.metas.handlingunits.picking.PickingCandidate;
+import de.metas.handlingunits.picking.PickingCandidateRepository;
+import de.metas.handlingunits.picking.PickingCandidateStatus;
+import de.metas.handlingunits.picking.requests.PickHURequest;
+import de.metas.inoutcandidate.api.IShipmentScheduleBL;
+import de.metas.inoutcandidate.api.IShipmentSchedulePA;
+import de.metas.inoutcandidate.api.ShipmentScheduleId;
+import de.metas.picking.api.PickingSlotId;
+import de.metas.quantity.Quantity;
+import de.metas.util.Services;
+import lombok.Builder;
+import lombok.NonNull;
+
+/*
+ * #%L
+ * de.metas.handlingunits.base
+ * %%
+ * Copyright (C) 2017 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+public class CreatePickingCandidatesCommand
+{
+	private final ITrxManager trxManager = Services.get(ITrxManager.class);
+	private final IShipmentScheduleBL shipmentScheduleBL = Services.get(IShipmentScheduleBL.class);
+	private final IShipmentSchedulePA shipmentSchedulesRepo = Services.get(IShipmentSchedulePA.class);
+	private final PickingCandidateRepository pickingCandidateRepository;
+
+	private final ShipmentScheduleId shipmentScheduleId;
+	private final HuId pickFromHuId;
+	private final PickingSlotId pickingSlotId;
+
+	@Builder
+	private CreatePickingCandidatesCommand(
+			@NonNull final PickingCandidateRepository pickingCandidateRepository,
+			@NonNull final PickHURequest request)
+	{
+		this.pickingCandidateRepository = pickingCandidateRepository;
+
+		this.shipmentScheduleId = request.getShipmentScheduleId();
+		this.pickFromHuId = request.getPickFromHuId();
+		this.pickingSlotId = request.getPickingSlotId();
+	}
+
+	public PickHUResult perform()
+	{
+		return trxManager.callInThreadInheritedTrx(this::performInTrx);
```

Comment:
Do we need `callInThreadInheritedTrx` here?
If we *really* need it, please add a little comment or javadoc about "why"

Reply:
Good piint! We do not need it!